File: Fragmentexercise1noappbar/app/src/main/java/com/cardlay/fragmentexercise1noappbar/MainActivity.java
package com.cardlay.fragmentexercise1noappbar;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.main_fragment, new WhiteFragment())
                    //.addToBackStack(null)
                    .commit();
        }

        Button btnWhite = findViewById(R.id.btnWhite);
        btnWhite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateFragment(new WhiteFragment());
            }
        });
        Button btnBlue = findViewById(R.id.btnBlue);
        btnBlue.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateFragment(new BlueFragment());
            }
        });
        Button btnGreen = findViewById(R.id.btnGreen);
        btnGreen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateFragment(new GreenFragment());
            }
        });
        Button btnRed = findViewById(R.id.btnRed);
        btnRed.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateFragment(new RedFragment());
            }
        });
        btnGreen.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateFragment(new GreenFragment());
            }
        });
    }

    private void updateFragment(Fragment currentFragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.main_fragment, currentFragment)
                .addToBackStack(null)
                .commit();
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
    }
}

File: Android 06 - Fragments/Fragmentexercise1noappbar/app/src/main/java/com/cardlay/fragmentexercise1noappbar/MainActivity.kt
package com.cardlay.fragmentexercise1noappbar

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.main_fragment, WhiteFragment()) //.addToBackStack(null)
                .commit()
        }
        val btnWhite = findViewById<Button>(R.id.btnWhite)
        btnWhite.setOnClickListener { updateFragment(WhiteFragment()) }
        val btnBlue = findViewById<Button>(R.id.btnBlue)
        btnBlue.setOnClickListener { updateFragment(BlueFragment()) }
        val btnGreen = findViewById<Button>(R.id.btnGreen)
        btnGreen.setOnClickListener { updateFragment(GreenFragment()) }
        val btnRed = findViewById<Button>(R.id.btnRed)
        btnRed.setOnClickListener { updateFragment(RedFragment()) }
        btnGreen.setOnClickListener { updateFragment(GreenFragment()) }
    }

    private fun updateFragment(currentFragment: Fragment) {
        supportFragmentManager
            .beginTransaction()
            .add(R.id.main_fragment, currentFragment)
            .addToBackStack(null)
            .commit()
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
    }
}